pkg/reconciler/openshift/clusterconfig_test.go

```diff
@@ -31,21 +34,26 @@ func makeTestPolicyRules() []rbacv1.PolicyRule {
 }
 
 func makeTestArgoCDForClusterConfig() *argoprojv1alpha1.ArgoCD {
-	allowedNamespaces := []string{testNamespace, "dummyNamespace"}
+	a := makeTestArgoCD()
+	a.Namespace = testNamespace
+	return a
+}
+
+func makeTestArgoCD() *argoprojv1alpha1.ArgoCD {
+	setClusterConfigNamespaces()
 	a := &argoprojv1alpha1.ArgoCD{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      testArgoCDName,
-			Namespace: testNamespace,
-		},
-		Spec: argoprojv1alpha1.ArgoCDSpec{
-			ManagementScope: argoprojv1alpha1.ArgoCDScope{
-				ClusterConfigNamespaces: allowedNamespaces,
-			},
+			Namespace: testDummyNameSpace,
 		},
 	}
 	return a
 }
 
+func setClusterConfigNamespaces() {
+	os.Setenv("ARGOCD_CLUSTER_CONFIG_NAMESPACES", "argocd,foo,bar")
```

Comment:
Couple of issues with this, you're setting it to the same value in each test, so you could set it once, but that's not what I'd do, I'd add a t.Cleanup function that could do `os.Unsetenv` _or_ restore the original value (probably the former).

But, there's not enough testing around the parsing of environment variable...
Try testing it with `argocd, foo, bar` for  example.

Reply:
I thought about this `argocd,  foo,  bar`, I was confused if I need to handle the spaces or should I say its not provided as per documentation.

Reply:
added a new function trimWhitespaces() to handle this.